watertap/costing/unit_models/clarifier.py

```diff
@@ -0,0 +1,316 @@
+        num_clarifier = math.ceil(
+            pyo.value(surface_area) / pyo.value(max_surface_area_limit)
+        )
+        blk.capital_cost_constraint = pyo.Constraint(
+            expr=blk.capital_cost
+            == pyo.units.convert(
+                num_clarifier
+                * (
+                    blk.costing_package.rectangular.concstruction_a_parameter
+                    * max_surface_area_limit**2
+                    + blk.costing_package.rectangular.concstruction_b_parameter
+                    * max_surface_area_limit
+                    + blk.costing_package.rectangular.concstruction_c_parameter
+                ),
+                to_units=blk.costing_package.base_currency,
+            )
+        )
+
+        blk.fixed_operating_cost_constraint = pyo.Constraint(
+            expr=blk.fixed_operating_cost
+            == pyo.units.convert(
+                num_clarifier
+                * (
+                    blk.costing_package.rectangular.O_and_M_a_parameter
+                    * max_surface_area_limit
+                    + blk.costing_package.rectangular.O_and_M_b_parameter
+                ),
+                to_units=blk.costing_package.base_currency
+                / blk.costing_package.base_period,
+            )
+        )
+
+    else:
+        blk.capital_cost_constraint = pyo.Constraint(
+            expr=blk.capital_cost
+            == pyo.units.convert(
+                blk.costing_package.rectangular.concstruction_a_parameter
+                * surface_area**2
+                + blk.costing_package.rectangular.concstruction_b_parameter
+                * surface_area
+                + blk.costing_package.rectangular.concstruction_c_parameter,
+                to_units=blk.costing_package.base_currency,
+            )
+        )
+
+        blk.fixed_operating_cost_constraint = pyo.Constraint(
+            expr=blk.fixed_operating_cost
+            == pyo.units.convert(
+                blk.costing_package.rectangular.O_and_M_a_parameter * surface_area
+                + blk.costing_package.rectangular.O_and_M_b_parameter,
+                to_units=blk.costing_package.base_currency
+                / blk.costing_package.base_period,
+            )
+        )
+
+
+def build_primary_clarifier_cost_param_block(blk):
+
+    blk.capital_a_parameter = pyo.Param(
+        initialize=120000 / 2776 * 12463,
+        doc="A parameter for capital cost",
+        units=pyo.units.USD_2021,
+    )
+
+    blk.capital_b_parameter = pyo.Param(
+        initialize=0.7,
+        doc="B parameter for construction cost",
+        units=pyo.units.dimensionless,
+    )
+
+
+@register_costing_parameter_block(
+    build_rule=build_primary_clarifier_cost_param_block,
+    parameter_block_name="primary",
+)
+def cost_primary_clarifier(blk, cost_electricity_flow=True):
+    """
+    Primary clarifier costing method
+    """
+    make_capital_cost_var(blk)
+
+    t0 = blk.flowsheet().time.first()
```

Comment:
Should `t0` and `cost_electricity_flow` also be included for the alternative costing methods? Not clear on whether the O&M costs include that already, but maybe you can clarify.

Reply:
Primary clarifier costing method is based on the `flow_vol` which has time index, but the other costing methods curently only depends on `surface_area` and does not have time index. And for circular and rectangular clarifier, the electricity has already been included in O&M cost so I don't register the electricity flow.

Reply:
Do we know what fraction of O&M is attributed to electricity?

Reply:
> Do we know what fraction of O&M is attributed to electricity? Yeah, from the paper it's around 3%

Yeah, from the paper it's around 3%

Reply:
I've revised this part for O&M costing, the electricity is registered separately, and the fixed O&M cost is adjusted based on the proportion.